ipa/ipa_utils.py

```diff
@@ -299,6 +299,11 @@ def get_test_files(test_dirs):
     tests = {}
     descriptions = {}
     for test_dir in test_dirs:
+        if not os.path.exists(test_dir):
+            raise IpaUtilsException(
+                'Test directory: %s not found.' % test_dir
+            )
+
```

Comment:
Should be a warning that we are skipping a given directory, proceed on a best effort basis.

Reply:
I know we discussed this before and chose to not throw an exception. In practice not having an explicit exception makes it harder to debug an issue.

Scenario:
- Supply a list of test directories with one or multiple invalid.
- If one of the chosen tests happens to be in the invalid test directory (most likely a typo), parse test files will raise an exception that the file does not exist.

This is a confusing exception because the actual problem is the directory does not exist. With an explicit directory check the exception would be the root of the problem rather than farther down the stack.

Reply:
OK, fair approach.